reference.md

```diff
@@ -1005,6 +1002,39 @@ macros into smaller scopes as well. Note that macros are a
 compile-time construct; they do not exist at runtime. As such macros
 cannot be exported at the bottom of a module like functions and other values.
 
+Broadly, a macro replaces its own expression with another.
+
+Macros should return a "quoted" expression (via `` ` ``) to be used as the
+replacement. Normally that expression would "unquote" a value (via `,`) but it
+does not have to. You may not "unquote" a value outside of a "quoted"
+expression.
+
+```fennel
+
+(macro print-even [n]
+  (if (= (% n 2) 0)
+    `(print ,n)))
```

Comment:
For this example to work, we would need to explicitly point out that `n` must be a compile-time literal. If you don't understand macros then you would expect `(print-even (+ 1 1))` to work, but it would fail to compile.

Reply:
Actually don't think I meant to push this one... :flushed: